src/instructlab/model/serve_backend.py

```diff
@@ -23,6 +25,38 @@ def signal_handler(num_signal, __):
     sys.exit(0)
 
 
+def get_gpu_count() -> int:
+    # Third Party
+    import torch
+
+    if torch.cuda.is_available():
+        return torch.cuda.device_count()
+    return 0
+
+
+def check_tensor_parallel_size(vllm_args: List) -> Tuple[bool, int, int]:
+    if contains_argument("--tensor-parallel-size", vllm_args):
+        # First Party
+        from instructlab.model.backends.vllm import get_argument
+
+        available_gpus = get_gpu_count()
+        effective_gpus = 0
```

Comment:
This doesn't seem to be quite the same of other uses of effective_gpus.  Other use cases I am aware of use it as the effective value between --gpus and tps.  This seems to only be tps.